packages/glimmer-runtime/lib/compiled/expressions/concat.ts

```diff
@@ -31,10 +31,28 @@ class ConcatReference extends CachedReference<string> {
   }
 
   protected compute(): string {
-    let parts = new Array<string>(this.parts.length);
+    let parts = new Array<string>();
```

Comment:
Yes, but at the end you join the elements with an empty string, so it doesn't matter if there is holes. Actually your implementation might also have holes.